vizro-core/src/vizro/managers/_data_manager.py

```diff
@@ -16,6 +16,26 @@
 pd_LazyDataFrame = Callable[[], pd.DataFrame]
 
 
+class VizroDataSet:
+    """A dataset with flask-caching config attached.
+
+    Examples:
+        >>> import plotly.express as px
+        >>> data_manager["iris"] = VizroDataSet(lambda: pd.DataFrame(), timeout=300)
+        >>> data_manager.__lazy_data["iris"]._cache_arguments == {"timeout": 300}
+    """
+    def __init__(self, data: pd_LazyDataFrame, timeout: Optional[int] = None, unless: Optional[Callable] = None):
+        self.data = data
+        self._cache_arguments: Dict[str, int] = {}
+        self.set_cache_config(timeout, unless)
+
+    def set_cache_config(self, timeout: int = None, unless: Callable = None):
+        """Sets the cache configuration for the dataset."""
+        self._cache_arguments["timeout"] = timeout
+        self._cache_arguments["unless"] = unless
+        logger.debug(f"set_cache_config: {self._cache_arguments}")
+
```

Comment:
Let's change this so that a user can directly do `data_manager["iris"]._cache_arguments = {"timeout": 10}` rather than having a `set_cache_config` function. That will be much nicer and more Pythonic.